restler/unit_tests/authentication_test_files/unit_test_server_auth_module.py

```diff
@@ -0,0 +1,4 @@
+def acquire_token(log):
```

Comment:
>log [](http://example.com/codeflow?start=18&length=3)

Should this also take data?

Reply:
Made a change to make data optional, please take a look at https://github.com/microsoft/restler-fuzzer/pull/666/commits/5e3466a80cee4ff9850add6829e69901e0e7f050 and let me know what you think

It probably makes sense to test module with data and without data? I can add another module test for this, can look into setting up a helper function for some of these auth tests as well

Reply:
I was thinking of fewer options for the user - having the function as required to take 'data' and 'log' arguments.  Then, 'data' could be directly passed from the settings if present (otherwise None), and 'log' would take None if no logging is specified.  What do you think?

Reply:
Sounds good to me. I need to update the documentation to reflect this as well, will push a commit later today

Reply:
Also, yes on your other points to factor out the common test logic into a helper and add a test for 'no data'.  The tests look great to me otherwise.

Reply:
Updated to pass auth logger by default, please let me know what you think

Reply:
This looks good

Reply:
split module test into data/no-data

modified all of the auth tests to use run_abc_smoke_test